File: week 1/pset/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Get a float number from user
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);
    
    // Convert float to int and then multiple by 100
    int cents = round(dollars * 100);
    int coins = 0;
    
    // Each while loop checks if the cents are more or equal than the value of each coin
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }
    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }
    while (cents >= 1)
    {
        cents -= 1;
        coins++;
    }
    // Prints the number of coins needed
    printf("Coins: %i\n", coins);
}

File: week 6/pset/cash/cash.py
def main():
    # Prompts user for a positive number
    while True:
        try:
            change = float(input("Change owed: "))
            if change >= 0:
                break
        # If the input is not a number, prints an error message
        except ValueError:
            print("Invalid input")

    # Multiplies the input by 100 and rounds it to the nearest integer
    cents = int(round(change * 100))
    coins = 0

    # Keep subtracting the change by the value of each coin and incrementing the counter
    while cents >= 25:
        cents -= 25
        coins += 1

    while cents >= 10:
        cents -= 10
        coins += 1

    while cents >= 5:
        cents -= 5
        coins += 1

    if cents == 1:
        coins += 1

    # Prints the amount of coins needed
    print(f"Coins: {coins}")


if __name__ == '__main__':
    main()